File: PartyApp/app/src/main/java/com/example/alex/partyapp/network/TesonetApi.java
package com.example.alex.partyapp.network;

import com.example.alex.partyapp.data.Account;
import com.example.alex.partyapp.data.Token;
import com.example.alex.partyapp.data.Server;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface TesonetApi {

    String API_URL = "http://playground.tesonet.lt/v1/";

    @POST("tokens")
    Call<Token> login(@Body Account account);

    @GET("servers")
    Call<List<Server>> getServers(@Header("Authorization") String authorization);
}

File: app/src/main/java/com/example/android_party/api/TesonetApi.kt
package com.example.android_party.api

import com.example.android_party.data.Server
import com.example.android_party.data.Token
import io.reactivex.Observable
import retrofit2.http.*

/**
 * The interface which provides methods to get result of webservices
 */
interface TesonetApi {
    /**
     * Get the auth token from the API
     */
    @POST("/v1/tokens?")
    fun getAuthToken(
        @Query("username") username: String,
        @Query("password") password: String
    ): Observable<Token>

    /**
     * Get ther servers list from the API
     */
    @GET("v1/servers")
    fun getServers(@Header("Authorization") authorization: String): Observable<List<Server>>
}